### ISSUE TITLE
How to use query strings?

### ISSUE THREAD

**Issue by joh-klein** [restify/clients#134]
looking through the code, I found this: https://github.com/restify/clients/blob/master/lib/HttpClient.js#L722

Evidently doing this is deprecated:
```
var options = {
  path: '/foo/bar',
  query: {
    foo: 'bar',
    abc: ['xyz', '123']
  }
};

client.get(options, function(err, req, res) { .. });
```

What is the accepted way then?

**Comment by DonutEspresso** [comment]
Hey @joh-klein, thanks for pointing this out! TBH, I'm not sure - that comment was before my time. I have been using it the way you have been here in this example, and I don't expect this change anytime soon.

**Comment by DonutEspresso** [comment]
@yunong maybe you may know?

**Comment by tanveerrafiqueIVL** [comment]
I also want to know how to use query strings? There's nothing regarding it in docs.

**Comment by DonutEspresso** [comment]
Hi @tanveerrafiqueIVL, you should continue to use `options.query` until further notice. We definitely need to get this in the docs, thanks for pointing this out. 👍

**Comment by tanveerrafiqueIVL** [comment]
So will it work with JSON Client too? Because it wasn't working for me.

**Comment by joh-klein** [comment]
For the JSONClient I am using "querystring".

So something like this:

```
var qs = require('querystring');
var params = {
    foo: 'bar',
    abc: ['xyz', '123']
};

var options = {
  path: '/foo/bar?' + qs.stringify(params)
};

client.get(options, function(err, req, res) { .. });
```

**Comment by DonutEspresso** [comment]
Apologies @tanveerrafiqueIVL, it looks like I remembered incorrectly. :(  I poked through the client code again, it looks like options are getting passed through to underlying [http.request](https://nodejs.org/api/http.html#http_http_request_options_callback) method. 

@joh-klein's approach is correct for now - you'll have to put it together yourself for the time being, though it appear we could make this easier by adding first class support for it.
